### PR TITLE
Add zoom support

### PR DESCRIPTION
**Pull request by mtrive**
**Problem**
Project Auditor often reports a lot of issues and depending on the screen resolution it might be tricky to read through the list and the issue details/recommendation.

**Solution**
With this PR, the user can easily adjust the font size via a zoom slider. Note that this is a user preference so it will not be saved in the Project Auditor asset in the project.

This video shows how it works: https://user-images.githubusercontent.com/12098182/106491599-f0aeb280-64ae-11eb-9ea2-5ffe06b06260.mp4

By no means this is the perfect solution, so let me know if you have alternative ideas.

### CODE REVIEW

**Review by unitystevem — COMMENTED** [review]
I'm not convinced that it's a good idea for individual Editor windows to start rolling their own solutions for text size. I think this needs to be fixed in the Editor as a whole rather than using an inconsistent piecemeal approach. However, setting that concern aside, this seems to be as good an implementation as any, so I'm not going to try to block it being merged in if there's a strong feeling that Project Auditor will really benefit from this. I just hope it can be removed again in future because a feature like this has been implemented for _all_ windows.

**Comment by mtrive** [comment]
> I'm not convinced that it's a good idea for individual Editor windows to start rolling their own solutions for text size. I think this needs to be fixed in the Editor as a whole rather than using an inconsistent piecemeal approach. However, setting that concern aside, this seems to be as good an implementation as any, so I'm not going to try to block it being merged in if there's a strong feeling that Project Auditor will really benefit from this. I just hope it can be removed again in future because a feature like this has been implemented for _all_ windows.

Thanks @unitystevem. I don't think it's a must-have feature since we lived without it until now, however, it does solve a real (niche?) problem. Having said that, I am in the process of checking whether there is already an editor-wide solution to this problem. If not, I think we should land it and re-evaluate if things change. It would be straightforward to remove.

File: Editor/UI/AnalysisView.cs [hunk 1]
```diff
@@ -192,23 +192,30 @@ public void OnGUI()
...
-
+            EditorGUILayout.BeginHorizontal();
             if (m_Desc.groupByDescription)
             {
-                EditorGUILayout.BeginHorizontal();
-                if (GUILayout.Button(Styles.CollapseAllButton, GUILayout.ExpandWidth(true), GUILayout.Width(100)))
+                if (GUILayout.Button(Styles.CollapseAllButton, EditorStyles.toolbarButton, GUILayout.ExpandWidth(true), GUILayout.Width(100)))
                     SetRowsExpanded(false);
-                if (GUILayout.Button(Styles.ExpandAllButton, GUILayout.ExpandWidth(true), GUILayout.Width(100)))
+                if (GUILayout.Button(Styles.ExpandAllButton, EditorStyles.toolbarButton, GUILayout.ExpandWidth(true), GUILayout.Width(100)))
```
Comment (berniegp) [inline Editor/UI/AnalysisView.cs:202]: If the intent is to keep this PR easily revertible when/if Unity supports font size changes, adding the `EditorStyles.toolbarButton` param should go in another PR.

File: Editor/UI/AnalysisView.cs [hunk 2]
```diff
@@ -192,23 +192,30 @@ public void OnGUI()
...
                     SetRowsExpanded(true);
-                EditorGUILayout.EndHorizontal();
             }
 
+            EditorGUILayout.Space();
+            EditorGUILayout.LabelField("Zoom: ", EditorStyles.toolbarButton, GUILayout.ExpandWidth(false), GUILayout.Width(80));
+            m_Preferences.fontSize = (int)GUILayout.HorizontalSlider(m_Preferences.fontSize, Preferences.k_MinFontSize, Preferences.k_MaxFontSize, GUILayout.ExpandWidth(false), GUILayout.Width(100));
+            m_Table.SetFontSize(m_Preferences.fontSize);
```
Comment (berniegp) [inline Editor/UI/AnalysisView.cs:209]: Another option would be to store 2 `GUIStyle`s (textarea and label) in `Preferences.cs` and update their attributes in a setter for `m_Preferences.fontSize`. It would avoid recreating them all the time too (not sure that really matters though)

File: Editor/UI/IssueTable.cs [hunk 3]
```diff
@@ -368,6 +384,16 @@ void CellGUI(Rect cellRect, TreeViewItem treeViewItem, int columnIndex, ref RowG
             if (rule != null && rule.severity == Rule.Severity.None) GUI.enabled = true;
         }
 
+        void CenterRectUsingSingleLineHeight(ref Rect rect)
```
Comment (berniegp) [inline Editor/UI/IssueTable.cs:387]: `CenterRectUsingSingleLineHeight` is defined in `IMGUI.Controls.TreeView` as non virtual. If overriding this method only when accessed through references of type `IssueTable` is the intent, use the `new` keyword to get rid of the warning. This does not make the method any more `virtual` however and is a bit of a code smell to me. Since this method is only called directly above in `CellGUI()`, I suggest renaming it to avoid confusion.

**Review by berniegp — CHANGES_REQUESTED** [review]
Feel free to ignore my change requests if they don't make sense :)
Thanks for implementing this btw! And I agree that a better approach would be for Unity to allow configurable font sizes, but that's probably a big change.

File: Editor/UI/IssueTable.cs (hunk 3 above)
Comment (mtrive) [reply Editor/UI/IssueTable.cs]: Good catch!
Since the new `CenterRectUsingSingleLineHeight` is almost identical to the base class one, I fixed it by adding the `new` keyword. For the reader's awareness I think I prefer this way.

File: Editor/UI/AnalysisView.cs (hunk 1 above)
Comment (mtrive) [reply Editor/UI/AnalysisView.cs:202]: You are absolutely right. However, in this case, I am not going to bother simply because there are more changes coming that will affect this code. Removing this feature will be simple enough anyway. Thanks!

**Review by berniegp — APPROVED** [review]
